WordPress/WordPressTest/AccountSettingsServiceTests.swift

```diff
@@ -71,4 +72,29 @@ class AccountSettingsServiceTests: CoreDataTestCase {
         }
         expect(self.managedAccountSettings()?.firstName).to(equal("Test"))
     }
+
+    func testCancelRefreshingSettings() throws {
+        // This test performs steps described in the link below to reproduce a crash.
+        // https://github.com/wordpress-mobile/WordPress-iOS/issues/20379#issuecomment-1481995663
+
+        let apiFired = expectation(description: "Get settings API fired")
+        stub(condition: isPath("/rest/v1.1/me/settings")) { _ in
+            apiFired.fulfill()
+            sleep(2)
+            return HTTPStubsResponse(jsonObject: [:], statusCode: 500, headers: nil)
+        }
+        service.refreshSettings()
+
+        // A 5 senconds timeout is used here becuase there is a 4 seconds delay of making the HTTP API request
+        // in the `refreshSettings` call above.
+        wait(for: [apiFired], timeout: 5)
```

Comment:
The 4 seconds delay in the test because of `sleep(2)` and the retry slows down the unit tests.

This is unfortunate. Unit tests should be as fast as we can make them.

I wonder if there's a way to test that the completion with status `failed` is called when the API request fails with  `URLError.cancelled` instead?

That would remove the need to simulate a delay in the networking at this level. 🤔 

I realize it'd be troublesome to write a new test and to insert probes for this behavior, but I'd love to avoid making the unit tests sleep if possible.

Reply:
I had another look at the code. It turns out I misread the code. The test can call `getSettingsAttempt` straight-away. With https://github.com/wordpress-mobile/WordPress-iOS/pull/20392/commits/b29c466f7ccfff3ae8b8f05df788ce1af730dde3, this unit test passes in just a bit over 0.5 seconds.

Reply:
BTW, the `sleep(2)` doesn't affect the test execution—the test finishes just at the time the sleep starts.

Reply:
`With https://github.com/wordpress-mobile/WordPress-iOS/commit/b29c466f7ccfff3ae8b8f05df788ce1af730dde3, this unit test passes in just a bit over 0.5 seconds.

🎉 

<img alt="image" src="https://user-images.githubusercontent.com/1218433/227852759-de57e764-2467-4f74-a93e-2cc2bb166926.png" width=300/>

> BTW, the `sleep(2)` doesn't affect the test execution—the test finishes just at the time the sleep starts.

Then, is the value in having it that delayed is to simulate network latency? How would you feel about adding a comment to explain that?

Reply:
My bad, the test code sure looks quite random. I've added [some comments](https://github.com/wordpress-mobile/WordPress-iOS/pull/20392/commits/970402334c16066d92f25edfdbd0e4eb93b7deee) to explain what's happening underneath.